// config/ajout.js
var crypto = require('crypto');
var rand = require('csprng');
var mongoose = require('mongoose');
var gravatar = require('gravatar');
var lieu = require('config/models');
var MongoClient = require('mongodb').MongoClient, assert = require('assert');
var url = 'mongodb://GBUTAPP:GBUTAPP1234@ds259499.mlab.com:59499/gbutapp';	
var ObjectId = require('mongodb').ObjectId; 

exports.ajoutmembre = function(nomM,prenomM,contactM,addresseM,anniversaireM,callback) {
	MongoClient.connect(url, function(err, db) {
	 db.collection("Membre").insert({nom:nomM,prenom:prenomM,contact:contactM,addresse:addresseM,anniversaire:anniversaireM},
	 	function(err){
			callback("true");
		});
	 db.close();
	});

}
exports.ajoutcellule = function(nomM,jourM,debutM,finM,lieuM,chefM,callback) {
	MongoClient.connect(url, function(err, db) {
	 db.collection("cellules").insert({nom:nomM,jour:jourM,debut:ISODate(debutM), fin: ISODate(finM),lieu:lieuM,chef:chefM},
	 	function(err){
			callback("true");
		});
	 db.close();
	});
}

exports.ajoutmvf = function(mois,theme,callback) {
	MongoClient.connect(url, function(err, db) {
	 db.collection("mvf").insert({mois:mois,theme:theme},
	 	function(err){
			callback("true");
		});
	 db.close();
	});
}

exports.ajoutebp = function(mois,libelle,callback) {
	MongoClient.connect(url, function(err, db) {
	 db.collection("texte").insert({mois:mois,libelle:libelle},
	 	function(err){
			callback("true");
		});
	 db.close();
	});
}


exports.ajoutpb = function(mois,groupe1,personnage1,groupe2,personnage2,groupe3,personnage3,callback) {
	MongoClient.connect(url, function(err, db) {
	 db.collection("pb").insert({mois:mois,groupe1:groupe1,groupe2:groupe2,groupe3:groupe3,personnage1:personnage1,personnage2:personnage2,personnage3:personnage3},
	 	function(err){
			callback("true");
		});
	 db.close();
	});
}

exports.ajoutcalendrier = function(date,libelle,callback) {
	MongoClient.connect(url, function(err, db) {
	 db.collection("programmes").insert({date:ISODate(date),libelle:libelle},
	 	function(err){
			callback("true");
		});
	 db.close();
	});
}

exports.ajoutgroupeeb = function(texte,date,idmembre,callback) {
	MongoClient.connect(url, function(err, db) {
	 db.collection("groupeeb").insert({texte:texte,date:ISODate(date),idmembre:idmembre},
	 	function(err){
			callback("true");
		});
	 db.close();
	});
}

exports.ajoutveillee = function(mois,libelle,callback) {
	MongoClient.connect(url, function(err, db) {
	 db.collection("veillee").insert({mois:mois,libelle:libelle},
	 	function(err){
			callback("true");
		});
	 db.close();
	});
}

exports.ajoutbiblio = function(mois,libelle,callback) {
	MongoClient.connect(url, function(err, db) {
	 db.collection("biblio").insert({mois:mois,libelle:libelle},
	 	function(err){
			callback("true");
		});
	 db.close();
	});
}

exports.ajoutrealisateurrg = function(mois,libelle,callback) {
	MongoClient.connect(url, function(err, db) {
	 db.collection("rg").insert({mois:mois,libelle:libelle},
	 	function(err){
			callback("true");
		});
	 db.close();
	});
}

exports.ajoutmenage = function(mois,libelle,callback) {
	MongoClient.connect(url, function(err, db) {
	 db.collection("menage").insert({mois:mois,libelle:libelle},
	 	function(err){
			callback("true");
		});
	 db.close();
	});
}
